In [1]:
import pandas as pd 

In [2]:

# Load the datasets
sales_df = pd.read_excel("SalesData.xlsx")
product_df = pd.read_excel("Product.xlsx")

In [3]:
# Merge sales and product data
merged_df = sales_df.merge(product_df, left_on="ProductKey", right_on="ID")

In [5]:
# Calculate total sales value
merged_df["TotalSales"] = merged_df["Quantity"] * merged_df["UnitPrice"]

In [7]:
# Sales distribution by product category
category_sales = merged_df.groupby("ProductCategory")["TotalSales"].sum().sort_values(ascending=False)

In [8]:
# Top product by total sales
top_product = merged_df.groupby("ProductName")["TotalSales"].sum().sort_values(ascending=False).head(1)

In [9]:
# Top category by total sales
top_category = category_sales.head(1)


In [10]:
# Display results
print("Sales Distribution by Product Category:\n", category_sales)
print("\nTop Product:\n", top_product)
print("\nTop Category:\n", top_category)

Sales Distribution by Product Category:
 ProductCategory
Food     16366811.55
Drink     1542420.94
Name: TotalSales, dtype: float64

Top Product:
 ProductName
Product 1968    746302.62
Name: TotalSales, dtype: float64

Top Category:
 ProductCategory
Food    16366811.55
Name: TotalSales, dtype: float64


---

### 📊 **Output Summary**

**Sales Distribution by Product Category:**

* 🍽️ **Food:** \$16,366,811.55
* 🥤 **Drink:** \$1,542,420.94

**Top Product:**

* 🏆 Product 1968 — \$746,302.62

**Top Category:**

* 🍽️ Food — \$16,366,811.55

---